File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return self.encode(s) == self.encode(t)
        
        
    def encode(self, s: str) -> str:
        b = [0] * 26
        for c in s:
            b[ord(c) - ord('a')] += 1
        r = ''
        for i in b:
            r += str(b) + ' '
        return r

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> arr (26, 0);

        if (s.size() != t.size()) return false;

        for (int i = 0; i < s.size(); i++) {
            arr[s[i] - 'a']++;
            arr[t[i] - 'a']--;
        }

        for (int i : arr) {
            if (i != 0) return false;
        }

        return true;
    }
};